tests/lib/external/snapd-testing-tools/utils/log-analyzer

```diff
@@ -1,63 +1,269 @@
 #!/bin/bash
 
 show_help() {
-    echo "usage: get-failed <PARSED-LOG>"
-    echo "       get-successful <PARSED-LOG>"
-    echo "       get-aborted <PARSED-LOG>"
-    echo "       get-total <PARSED-LOG>"
-    echo "       list-failed <PARSED-LOG>"
-    echo "       check-rexecution <PARSED-LOG>"
+    echo "usage: log-analyzer list-failed-tasks <EXEC-PARAM> <PARSED-LOG>"
+    echo "       log-analyzer list-executed-tasks <EXEC-PARAM> <PARSED-LOG>"
+    echo "       log-analyzer list-successful-tasks <EXEC-PARAM> <PARSED-LOG>"
+    echo "       log-analyzer list-aborted-tasks <EXEC-PARAM> <PARSED-LOG>"
+    echo "       log-analyzer list-all-tasks <EXEC-PARAM>"
+    echo "       log-analyzer list-reexecute-tasks <EXEC-PARAM> <PARSED-LOG>"
     echo ""
     echo "Get general information about the current system"
+    echo ""
+    echo "Available options:"
+    echo "  -h --help   show this help message."
+    echo ""
+    echo "COMMANDS:"
+    echo "  list-failed-tasks        list the tasks that failed during execute"
+    echo "  list-executed-tasks      list the tasks that were executed"
+    echo "  list-successful-tasks    list the successful tasks"
+    echo "  list-aborted-tasks       list the aborted tasks (needs spread to be installed)"
+    echo "  list-all-tasks           list all the tasks"
+    echo "  list-reexecute-tasks     list the tasks to re-execute to complete (includes aborted and failed tasks)"
+    echo ""
+    echo "PARSED-LOG: This is the output generated by the log-parser tool"
+    echo "EXEC-PARAM: this is the parameter used to run spread (something like this BACKEND:SYSTEM:SUITE)"
+    echo ""
 }
 
-get_successful() {
+_check_log() {
     local log="$1"
-    TESTS_SUCCESSFUL=$(jq -r '.[] | select( .type == "result") | ( select( .result_type == "Successful") | .number ) // 0 ' "$log" | awk '{s+=$1} END {print s}')
+
+    if [ -z "$log" ]; then
+        echo "log.analyzer: the log file cannot be empty"
+        exit 1
+    elif [ ! -f "$log" ]; then
+        echo "log.analyzer: the log file $log does not exist"
```

Comment:
the problem with this is that we have the same behavior in all the tools, perhaps it could be included in a following pr and include other tools as well, does it make sense?